File: Project5/haskell/task2.hs
{- Memory management in Haskell

Svita Kiran
CS333
11/13/2025
-}

module Main where

makeLargeList :: Int -> Int
makeLargeList n =
    let bigList = [1..n]
    in length bigList

stressMemory :: Int -> IO ()
stressMemory 0 = putStrLn "done"
stressMemory n = do
    let val = makeLargeList 500000
    val `seq` return ()
    stressMemory (n - 1)

main :: IO ()
main = do
    putStrLn "memory stress test"
    stressMemory 5

File: Project5/go/task2.go
/*
 * Memory management in Go
 * 
 * Svita Kiran
 * 11/13/2025
 * CS333
*/

package main

import "fmt"

type Node struct {
    Value int
    Next  *Node
}

func makeGarbage() {
    head := &Node{1, nil}
    for i := 2; i <= 10; i++ {
        head = &Node{i, head}
    }

    head = nil
}

func main() {
    fmt.Println("creating garbage")
    for i := 0; i < 5; i++ {
        makeGarbage()
    }
    fmt.Println("done and garbage collection will run automatically")
}
